worlds/mmbn3/__init__.py

```diff
@@ -0,0 +1,458 @@
+                    entrance.access_rule = lambda state: state.has(ItemName.PETCase, self.player)
+
+                # ACDC Pending with Start Randomizer
+                # if connection == RegionName.ACDC_Cyberworld:
+                #     entrance.access_rule = lambda state: state.has(ItemName.CACDCPas, self.player)
+                if connection == RegionName.SciLab_Cyberworld:
+                    entrance.access_rule = lambda state: \
+                        state.has(ItemName.CSciPas, self.player) or \
+                        state.can_reach(RegionName.SciLab_Overworld, "Region", self.player)
+                if connection == RegionName.Yoka_Cyberworld:
+                    entrance.access_rule = lambda state: \
+                        state.has(ItemName.CYokaPas, self.player) or \
+                        (
+                            state.can_reach(RegionName.SciLab_Overworld, "Region", self.player) and
+                            state.has(ItemName.Press, self.player)
+                        )
+                if connection == RegionName.Beach_Cyberworld:
+                    entrance.access_rule = lambda state: state.has(ItemName.CBeacPas, self.player) and\
+                        state.can_reach(RegionName.Yoka_Overworld, "Region", self.player)
+
+                if connection == RegionName.Undernet:
+                    entrance.access_rule = lambda state: self.explore_score(state) > 8 and\
+                        state.has(ItemName.Press, self.player)
+                if connection == RegionName.Secret_Area:
+                    entrance.access_rule = lambda state: self.explore_score(state) > 12 and\
+                        state.has(ItemName.Hammer, self.player)
+                if connection == RegionName.WWW_Island:
+                    entrance.access_rule = lambda state:\
+                        state.has(ItemName.Progressive_Undernet_Rank, self.player, 8)
+                region.exits.append(entrance)
+
+    def create_items(self) -> None:
+        # First add in all progression and useful items
+        required_items = []
+        for item in all_items:
+            if item.progression != ItemClassification.filler:
+                freq = self.item_frequencies[item.itemName] if item.itemName in self.item_frequencies else 1
+                required_items += [item.itemName] * freq
+
+        for itemName in required_items:
+            self.multiworld.itempool.append(self.create_item(itemName))
+
+        # Then, get a random amount of fillers until we have as many items as we have locations
+        filler_items = []
+        for item in all_items:
+            if item.progression == ItemClassification.filler:
+                freq = player_item_frequencies[self.player][item.itemName] \
+                    if self.player in player_item_frequencies and item.itemName in player_item_frequencies[self.player] \
+                    else item_frequencies[item.itemName] if item.itemName in item_frequencies else 1
```

Comment:
The item frequencies for the filler items is the same as the others, it's how many items there are in the game, but some of them won't end up fitting since their locations have been replaced with things like the progressive undernet ranks that normally come from story progress instead of item checks. So it's more like it's randomly deciding which filler items to not include in the generation. So, if a filler item has a frequency of 4, that means there are four of them in the pool that all _could_ get in, but might not. I see now that I never actually remove them from the list if they're chosen, so I'm adding in that functionality now, it should be have more like a normal item now.